### PR TITLE
[#182, Part 4] Add functions for fitting parabolic and exponential functions.

### PR DESCRIPTION
**Pull request by adamcw**
For #182. Add least-squares fitting functions for both parabolic and exponential functions. To be used by the Grapher to fit the plots.

### CODE REVIEW

File: client-js/app/scripts/fit.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+ *
+ * Citation:
+ * Weisstein, Eric W. "Least Squares Fitting--Polynomial."
+ * From MathWorld--A Wolfram Web Resource.
+ * http://mathworld.wolfram.com/LeastSquaresFittingPolynomial.html
+ */
+export function fitParabola(data: number[][],
+                            xMin: number,
+                            xMax: number,
+                            samples: number = 1000) {
```
Comment (maffoo) [inline client-js/app/scripts/fit.ts]: Add explicit return type.

File: client-js/app/scripts/fit.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+ *
+ * Citation:
+ * Weisstein, Eric W. "Least Squares Fitting--Exponential."
+ * From MathWorld -- A Wolfram Web Resource.
+ * http://mathworld.wolfram.com/LeastSquaresFittingExponential.html
+ */
+export function fitExponential(data: number[][],
+                               xMin: number,
+                               xMax: number,
+                               samples: number = 1000) {
```
Comment (maffoo) [inline client-js/app/scripts/fit.ts]: Add explicit return type.

File: client-js/app/scripts/fit.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    sumX2Y += Math.pow(x, 2) * y;
+    sumYLogY += y * Math.log(y);
+    sumXY += x * y;
+    sumXYLogY += x * y * Math.log(y);
+    sumY += y;
+  }
+
+  let [A, B] = [NaN, NaN];
+  const fittedExponential = [];
+  const denominator = (sumY * sumX2Y - Math.pow(sumXY, 2));
```
Comment (maffoo) [inline client-js/app/scripts/fit.ts]: Could call this `determinant` as above, since this is also just computing a matrix inverse.

**Comment by maffoo** [comment]
A few small comments, otherwise LGTM. Note that the build issue is addressed in #270.

**Comment by adamcw** [comment]
Merged this into Part 1 manually.

**Pull request closed without merging**